core/compiler.cpp

```diff
@@ -138,10 +138,16 @@ partitioning::GraphAndMapping BuildHybridGraph(
     torch::jit::Block* block,
     CompileSpec cfg,
     ir::StaticParams static_params,
-    ir::CollectionTypeMap first_use_types) {
+    ir::CollectionTypeMap first_use_types,
+    bool expect_full_compilation = false) {
   auto convert_info = cfg.convert_info;
   auto partitioning_info = cfg.partitioning_info;
 
+  // Any nonzero block size is valid if full compilation to TRT is desired
+  if (expect_full_compilation) {
+    partitioning_info.min_block_size = 1;
```

Comment:
I don't quite understand this, if you expect full compilation, shouldnt the min block size be the full graph? min_block_size 1 would mean any singular layer can be by itself in trt

Reply:
The purpose of this was to override the default of `min_block_size=3` in partitioning. I agree that the min block size should be the full graph, but if the number of non-collection operations in the graph is less than 3, this partitioning phase would not respect full compilation (it would cause those operations to be executed in Torch). This condition is a way around that issue - by allowing any size of TRT block, we allow any size of TRT graph to be valid in partitioning.

Reply:
min_block_size only looks at the number of contiguous TRT operations. If this is an issue then it would probably effect non full compilation cases as well. There should be explicit handling of graphs that are smaller than `min_block_size` to just lower that threshold to 1, but that should probably be in partitioning and not specific to this use case.

Reply:
I see - I agree that this should probably be in partitioning. I am wondering what your thoughts are on adding `expect_full_compilation` as an argument to `partitioning::partition`?

I added a commit that has a test case to elicit the specific behavior that I want to catch with this condition. Additionally, I have moved the condition into a narrower scope, closer to where it is used so that the code is more understandable (though still in `compiler.cpp`, for now).